legion/engine/core/scenemanagement/scenemanager.cpp

```diff
@@ -1,69 +1,124 @@
-#include <core/scenemanagement/scene.hpp>
+#include <core/scenemanagement/components/scene.hpp>
 #include <core/serialization/serializationUtil.hpp>
+#include <core/logging/logging.hpp>
+#include <core/common/string_extra.hpp>
+#include <core/defaults/defaultcomponents.hpp>
+//#include <rendering/components/camera.hpp>
 
 
 namespace legion::core::scenemanagement
 {
+    int SceneManager::sceneCount;
+    std::string SceneManager::currentScene = "Main";
     std::unordered_map < id_type, std::string> SceneManager::sceneNames;
     std::unordered_map<id_type, ecs::component_handle<scene>> SceneManager::sceneList;
+    std::atomic_bool SceneManager::doNotCreateEntities{ false };
 
+    ecs::entity_handle SceneManager::create_scene_entity()
+    {
+        ecs::entity_handle sceneEntity;
+        static ecs::EntityQuery sceneEntities = m_ecs->createQuery<scenemanagement::scene>();
+        sceneEntities.queryEntities();
+        if (sceneEntities.size() == 0)
+        {
+            log::debug("Creating a Scene Entity");
+            sceneEntity = m_ecs->createEntity();
+            sceneEntity.add_component<hierarchy>();
+            sceneEntity.add_component<scene>();
+            std::vector<ecs::entity_handle> children;
+            for (size_type i = 0; i < m_ecs->world.child_count(); i++)
+            {
+                children.push_back(m_ecs->world.get_child(i));
+            }
+            log::debug("Child count {}",m_ecs->world.child_count());
```

Comment:
some debug logs here, if you still need them you can leave them in, but remove them if possible

Reply:
I wanna keep them in until I am more confident in scenemanagment.